redfish-core/lib/service_root.hpp

```diff
@@ -26,30 +26,71 @@ namespace redfish
 inline void
     handleACFWindowActive(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
 {
+    // Redfish property ACFWindowActive = true when D-Bus property
+    // allow_unauth_upload is true (Redfish property AllowUnauthACFUpload).
     crow::connections::systemBus->async_method_call(
         [asyncResp](const boost::system::error_code ec,
-                    const std::variant<bool>& retVal) {
+                    const std::variant<bool>& allowed) {
             if (ec)
             {
-                BMCWEB_LOG_ERROR << "Failed to read ACFWindowActive property";
-                // Default value when panel app is unreachable.
-                asyncResp->res.jsonValue["Oem"]["IBM"]["ACFWindowActive"] =
-                    false;
+                BMCWEB_LOG_ERROR
+                    << "D-Bus response error reading allow_unauth_upload: "
+                    << ec;
+                messages::internalError(asyncResp->res);
                 return;
             }
-            const bool* isACFWindowActive = std::get_if<bool>(&retVal);
-            if (isACFWindowActive == nullptr)
+            const bool* allowUnauthACFUpload = std::get_if<bool>(&allowed);
+            if (allowUnauthACFUpload == nullptr)
             {
-                BMCWEB_LOG_ERROR << "nullptr for ACFWindowActive";
+                BMCWEB_LOG_ERROR << "nullptr for allow_unauth_upload";
                 messages::internalError(asyncResp->res);
                 return;
             }
-            asyncResp->res.jsonValue["Oem"]["IBM"]["ACFWindowActive"] =
-                *isACFWindowActive;
+            if (*allowUnauthACFUpload == true)
+            {
+                asyncResp->res.jsonValue["Oem"]["IBM"]["ACFWindowActive"] =
```

Comment:
Shouldn't this be the new property AllowUnauthACFUpload ?

Reply:
No.  I'll change the comments to make this more clear:
> Redfish property ACFWindowActive=true when either of these is true:
> - D-Bus property allow_unauth_upload.  (This is aka the Redfish property AllowUnauthACFUpload which the BMC admin can control.)
> - D-Bus property ACFWindowActive.  (This is aka the Redfish property ACFWindowActive under /redfish/v1/AccountService/Accounts/service property Oem.IBM.ACF.  The value is provided by the PanelApp and is true when panel function 74 is active.)
>
> Check D-Bus property allow_unauth_upload first.